lib/ansible/modules/windows/win_share.ps1

```diff
@@ -111,26 +111,34 @@ Function NormalizeAccounts
 
 $result = @{
     changed = $false
+    actions = @() # More for debug purposes
 }
 
-$params = Parse-Args $args
+$params = Parse-Args $args -supports_check_mode $true
+$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
 
 $name = Get-AnsibleParam -obj $params -name "name" -type "str" -failifempty $true
 $state = Get-AnsibleParam -obj $params -name "state" -type "str" -default "present" -validateset "present","absent"
 
+if (-not (Get-Command -Name Get-SmbShare -ErrorAction SilentlyContinue)) {
+    Fail-Json $result "The current host does not support the -SmbShare cmdlets required by this module. Please run on Server 2012 or Windows 8 and later"
+}
+
 Try {
-    $share = Get-SmbShare $name -ErrorAction SilentlyContinue
+    $share = Get-SmbShare -Name $name -ErrorAction SilentlyContinue
     If ($state -eq "absent") {
         If ($share) {
-            Remove-SmbShare -Force -Name $name
+            if (-not $check_mode) {
+                Remove-SmbShare -Force -Name $name
+            }
+            $result.actions += "Remove-SmbShare -Force -Name $name"
             $result.changed = $true
         }
-    }
-    Else {
+    } Else {
         $path = Get-AnsibleParam -obj $params -name "path" -type "path" -failifempty $true
         $description = Get-AnsibleParam -obj $params -name "description" -type "str" -default ""
 
-        $permissionList = Get-AnsibleParam -obj $params -name "list" -type "bool" -default "no" -validateset "no","yes" -resultobj $result
+        $permissionList = Get-AnsibleParam -obj $params -name "list" -type "bool" -default $false
```

Comment:
I agree this should be at the same section as the other parameters but I didn't want to chop up the module in one PR. Just focused on the testing and check mode support before trying to simplify it.